tests/test_internal.py

```diff
@@ -207,3 +209,22 @@ def test_schema_is_valid():
         collect_invalid_schemas(cs.nullable_schema(cs.int_schema(metadata={HAS_INVALID_SCHEMAS_METADATA_KEY: True})))
         is True
     )
+
+
+@pytest.mark.parametrize(
+    'decimal,normalize,decimal_places,digits',
+    [
+        (Decimal('0.0'), False, 1, 1),
+        (Decimal('0.0'), True, 0, 1),
+        (Decimal('0.000'), False, 3, 3),
+        (Decimal('0.000'), True, 0, 1),
+        (Decimal('0.0001'), False, 4, 4),
+        (Decimal('0.0001'), True, 4, 4),
+        (Decimal('123.123'), False, 3, 6),
+        (Decimal('123.123'), True, 3, 6),
+        (Decimal('123.1230'), False, 4, 7),
+        (Decimal('123.1230'), True, 3, 6),
```

Comment:
```suggestion
        (Decimal('123.1230'), 4, 7),
        (Decimal('123.123'), 3, 6),
```

and so on for the other cases looks easier to read